app/components/metadata-panel.tsx

```diff
@@ -196,9 +201,12 @@ export const OpenMetadataPanelWrapper = ({
   children: React.ReactNode;
 }) => {
   const classes = useOtherStyles();
-  const openDimenion = useMetadataPanelStore((state) => state.openDimension);
+  const store = React.useContext(MetadataPanelStoreContext);
+  const { openDimension } = useStore(store, (state) => ({
+    openDimension: state.openDimension,
+  }));
```

Comment:
We could only have the openDimension returned here, here since we always return a new object from the state, we will get unnecessary updates. See https://tkdodo.eu/blog/working-with-zustand#prefer-atomic-selectors